skins/oasis/modules/UserPagesHeaderController.class.php

```diff
@@ -13,56 +13,59 @@ class UserPagesHeaderController extends WikiaController {
 	public function init() {
 		$this->content_actions = $this->app->getSkinTemplateObj()->data['content_actions'];
 		$this->isUserProfilePageExt = false;
-		$this->actionMenu = array();
+		$this->actionMenu = [ ];
 		$this->comments = null;
 		$this->editTimestamp = null;
 	}
 
 	/**
 	 * Checks whether given user name is the current user
+	 * @param string $userName User name to check against the current user
+	 * @return bool Whether the current user matches the given username
 	 */
 	public static function isItMe( $userName ) {
-		global $wgUser;
+		$wgUser = F::app()->wg->User;
 		return $wgUser->isLoggedIn() && ( $userName == $wgUser->getName() );
 	}
 
 	/**
-	 * Get name of the user this page referrs to
+	 * Get name of the user this page refers to
+	 * @param Title $title Title of the current page
+	 * @param array $namespaces User/user talk/blog namespaces
+	 * @param bool $fallbackToGlobal Whether to use the username from $wgUser as fallback
+	 * @return string User name this page refers to
 	 */
 	public static function getUserName( Title $title, $namespaces, $fallbackToGlobal = true ) {
 		wfProfileIn( __METHOD__ );
-		global $wgUser, $wgRequest;
+		$wg = F::app()->wg;
 
 		$userName = null;
 
-		if ( in_array( $title->getNamespace(), $namespaces ) ) {
+		if ( $title->inNamespaces( $namespaces ) ) {
 			// get "owner" of this user / user talk / blog page
-			$parts = explode('/', $title->getText());
-		}
-		else if( $title->getNamespace() == NS_SPECIAL ) {
-				if( $title->isSpecial( 'Following' ) || $title->isSpecial( 'Contributions' )) {
-					$target = $wgRequest->getText('target');
-					if( $target != '' ) {
-						// /wiki/Special:Contributions?target=FooBar (RT #68323)
-						$parts = array( $target );
-					}
-					else {
-						// get user this special page referrs to
-						$parts = explode( '/', $wgRequest->getText( 'title', false ) );
-
-						// remove special page name
-						array_shift( $parts );
-					}
+			$parts = explode( '/', $title->getText() );
+		} else if ( $title->isSpecialPage() ) {
+			if ( $title->isSpecial( 'Following' ) || $title->isSpecial( 'Contributions' ) ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1066)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Following" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1192)